### ISSUE TITLE
multiple pod CIDR ranges - Non RFC 1918 IP addresses

### ISSUE THREAD

**Issue by JoshStead** [Azure/AKS#2107]
Hello

I'm trying to get an answer to understand if it is possible to replicate a feature that is possible in EKS where you can assign and use the network ranges 100.64.0.0/10 and 198.19.0.0/16 so that pods don't consume any RFC 1918 IP addresses in your VNet when using the CNI network plugin (VPC in AWS).

This link probably explains it better: https://aws.amazon.com/premiumsupport/knowledge-center/eks-multiple-cidr-ranges/

Below is an extract from this guide:

_Note: If you run your pods on different CIDR ranges, then you get more available IP addresses for pods managed by Amazon EKS and more flexibility for your networking architectures. If you add secondary CIDR blocks to a VPC from the 100.64.0.0/10 and 198.19.0.0/16 ranges and use CNI custom networking, then your pods don't consume any RFC 1918 IP addresses in your VPC._

Thanks in advance

**Comment by paulgmiller** [comment]
Different node pools in AKS can use different subnets with differetn ip ranges. PLease note the limitations carefully. I am not sure if that's what you're asking though.
: https://docs.microsoft.com/en-us/azure/aks/use-multiple-node-pools#add-a-node-pool-with-a-unique-subnet-preview

If you're using kubenet you're restricted to one pod cidr.

**Comment by paulgmiller** [comment]
Sorry I missed the "non" before rfc 1918. While I believe you can put nodes and pods on a subnet using non private ips just by brining a subnet swith that cidr, doing so can be imprudent as you'll have a hard time routing to the acutal public ip space if ever needed. 

Any context on why you want to use non private ips?

**Comment by poncin** [comment]
If I may, let me reformulate the initial question.
Are RFC6598 IP ranges https://datatracker.ietf.org/doc/html/rfc6598 supported on AKS, like on AWS EKS ?
NON-RFC1918 does not mean "public IP".
Context is RFC-1918 being already allocated and saturated in large enterprises (add a couple of M&A and you don't have much available without full re-IP), not leaving much room for AKS.

**Comment by JoshStead** [comment]
Hi @paulgmiller 

I asked this question early on my AKS journey where my company were looking to build a reusable container solution/product that will be used within my company and where possible replicate what had been done already on EKS. 

One area we found was different was that EKS had been configured to use 100.64.0.0/10 and 198.19.0.0/16 (NON-RFC1918 ip ranges) for pods and use a private address IP range to run the worker nodes e.g 10.10.1.0/25 with CNI networking. This would be fine using kubenet on AKS but not Azure CNI. 

We have decided to stay with Azure CNI and it's understood the ip addressing will be different between the two cloud providers.
